pkgs/games/mudlet/default.nix

```diff
@@ -1,46 +1,72 @@
-{ fetchurl, unzip, stdenv, makeWrapper, qtbase, yajl, libzip, hunspell
-, boost, lua5_1, luafilesystem, luazip, lrexlib-pcre, luasql-sqlite3, qmake }:
+{ fetchFromGitHub, fetchpatch, unzip, stdenv, qt5, pcre, pugixml, qtbase, qtmultimedia, qttools, yajl, libzip, hunspell
+, boost, libGLU, lua5_1, luafilesystem, luautf8, luazip, lrexlib-pcre, luasql-sqlite3, lua-yajl, cmake, qmake, which, }:
 
 stdenv.mkDerivation rec {
   pname = "mudlet";
-  version = "3.0.0-delta";
+  version = "4.0.3";
 
-  src = fetchurl {
-    url = "https://github.com/Mudlet/Mudlet/archive/Mudlet-${version}.tar.gz";
-    sha256 = "08fhqd323kgz5s17ac5z9dhkjxcmwvcmvhzy0x1vw4rayhijfrd7";
+  src = fetchFromGitHub {
+    owner = "Mudlet";
+    repo = "Mudlet";
+    rev = "Mudlet-${version}";
+    fetchSubmodules = true;
+    sha256 = "18bl4k0qgh47d9k5ipfvypfj1il678c0ws64a8adn8k21jajzkik";
   };
 
-  nativeBuildInputs = [ makeWrapper qmake ];
+  patches = [
+    ( fetchpatch {
+      url = "https://github.com/abbradar/Mudlet/commit/4f9450e9f66ac72f502e52951ff04b8ac8d123b3.patch";
+      name = "hunspell-1.7";
+      sha256 = "09qggls4pzpd8h9h10fbpfd7x3kr7fjp9axdwz98igpwy714n98j";
+    })
+  ];
+
+  nativeBuildInputs = [ cmake qt5.wrapQtAppsHook qttools which ];
   buildInputs = [
-    unzip qtbase lua5_1 hunspell libzip yajl boost
-    luafilesystem luazip lrexlib-pcre luasql-sqlite3
+    unzip pcre pugixml qtbase qtmultimedia lua5_1 libzip libGLU yajl boost
+    luafilesystem luazip lrexlib-pcre luasql-sqlite3 lua-yajl hunspell luautf8
   ];
 
-  preConfigure = "cd src";
+  WITH_FONTS = false;
+  WITH_UPDATER = "NO";
+
+  preConfigure = ''
+    export LUA_PATH="$NIX_LUA_PATH"
+    export LUA_CPATH="$NIX_LUA_CPATH"
+  '';
+
+  enableParallelBuilding = true;
 
   installPhase = let
     luaZipPath = "${luazip}/lib/lua/5.1/?.so";
     luaFileSystemPath = "${luafilesystem}/lib/lua/5.1/?.so";
     lrexlibPath = "${lrexlib-pcre}/lib/lua/5.1/?.so";
     luasqlitePath = "${luasql-sqlite3}/lib/lua/5.1/?.so";
+    luayajlPath = "${lua-yajl}/lib/lua/5.1/?.so";
+    luaUtf8Path = "${luautf8}/lib/lua/5.1/?.so";
   in ''
     mkdir -pv $out/bin
-    cp mudlet $out
-    cp -r mudlet-lua $out
+    cp src/mudlet $out
+    mkdir -pv $out/share/mudlet
+    cp -r ../src/mudlet-lua/lua $out/share/mudlet/
+
+    mkdir -pv $out/share/applications
+    cp ../mudlet.desktop $out/share/applications/
 
-    makeWrapper $out/mudlet $out/bin/mudlet \
-      --set LUA_CPATH "${luaFileSystemPath};${luaZipPath};${lrexlibPath};${luasqlitePath}" \
+    mkdir -pv $out/share/pixmaps
+    cp -r ../mudlet.png $out/share/pixmaps/
+
+    makeQtWrapper $out/mudlet $out/bin/mudlet \
+      --set LUA_CPATH "${luaFileSystemPath};${luaZipPath};${lrexlibPath};${luasqlitePath};${luayajlPath};${luaUtf8Path}" \
+      --prefix LUA_PATH : "$NIX_LUA_PATH" \
```

Comment:
have you tried using an environment  instead? `luaEnv = lua.withPackages(p: [luaFileSystemPath ...])`. Will make the path simpler and indenpendant of the lua version, like one could override with lua52 and still have the paths valid instead of `luaUtf8Path = "${luautf8}/lib/lua/5.1/?.so";`

Reply:
I didn't because I didn't know about that method. You are right. It's way cleaner.
I couldn't get rid of all dependencies on the lua version, but I moved that version in a variable at the top for easier maintenance and discovery.

Reply:
Scratch that. Got it working by overwriting the lua version in `all-packages.nix`.